apps/protocol-frontend/src/hooks/useDaoUserCreate.ts

```diff
@@ -0,0 +1,44 @@
+import { useUser } from '../contexts/UserContext';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { DaoUserCreateValues } from '../types/forms';
+import { useGovrnToast } from '@govrn/protocol-ui';
+
+export const useDaoUserCreate = () => {
+  const toast = useGovrnToast();
+  const { govrnProtocol: govrn } = useUser();
+  const queryClient = useQueryClient();
+
+  const { mutateAsync, isLoading, isError, isSuccess } = useMutation(
+    async (newDaoUser: DaoUserCreateValues) => {
+      const mutationData = await govrn.guild.user.create({
+        data: {
+          userId: newDaoUser.userId,
+          guildId: newDaoUser.guildId,
+        },
+      });
+      return mutationData;
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries(['activityTypes']); // invalidate the activity types query -- covers all args
+        queryClient.invalidateQueries(['userDaos']); // invalidate the userDaos query -- covers all args
+        queryClient.invalidateQueries(['contributionList']);
+        queryClient.invalidateQueries(['contributionInfiniteList']);
+        queryClient.invalidateQueries(['ContributionGetCountYear']);
+        queryClient.invalidateQueries(['contributionGetCount']);
```

Comment:
Why are we invalidating all these contribution queries? Shouldn't this really be userDaos only?

Reply:
I was going back and forth about this. The reason I landed on invalidating these is because they'd be impacted by having a user have a new DAO added to their list. I wasn't completely confident it would be invalidated if we only did the user's DAOs.